lib/v2/screens/app/interactor/usecases/get_initial_deep_link.dart

```diff
@@ -1,27 +1,52 @@
 import 'package:async/async.dart';
+
 // ignore: import_of_legacy_library_into_null_safe
 import 'package:seeds/features/scanner/telos_signing_manager.dart';
 import 'package:seeds/v2/datasource/local/settings_storage.dart';
-import 'package:uni_links/uni_links.dart';
 
 class GetInitialDeepLinkUseCase {
-  Future<Result> run() {
-    return getInitialLink()
-        .then((link) => _getRequestData(link))
-        // ignore: return_of_invalid_type_from_catch_error
-        .catchError((onError) => ErrorResult(onError));
-  }
+  Future<Result> run(Uri newLink) async {
+    var splitUri = newLink.query.split('=');
+    var placeHolder = splitUri[0];
+    var esrUrl = splitUri[1];
+
+    SeedsESR request = SeedsESR(uri: esrUrl);
 
-  Future<Result> _getRequestData(String? link) {
-    if (link != null) {
-      SeedsESR request = SeedsESR(uri: link);
-      return request
-          .resolve(account: settingsStorage.accountName)
-          .then((value) => ValueResult(value))
-          // ignore: return_of_invalid_type_from_catch_error
-          .catchError((onError) => ErrorResult(onError));
+    await request.resolve(account: settingsStorage.accountName);
+    var action = request.actions.first;
+    var data = Map<String, dynamic>.from(action.data as Map<dynamic, dynamic>);
+
+    var deepLinkPlaceHolder = DeepLinkPlaceHolder.LINK_UNKNOWN;
```

Comment:
Capital letters in enum options confuse me, they look like more classes 🤨

Reply:
> Use lowerCamelCase for constant variables, including enum values.

https://dart.dev/guides/language/effective-dart/style#prefer-using-lowercamelcase-for-constant-names

Reply:
agree :)